service/router.go

```diff
@@ -67,6 +67,10 @@ func AdminRouter() *mux.Router {
 	router.Handle("/v1/keypairs/{id:[0-9]+}/enable", Middleware(http.HandlerFunc(KeypairEnableHandler))).Methods("POST")
 	router.Handle("/v1/keypairs/assertion", Middleware(http.HandlerFunc(KeypairAssertionHandler))).Methods("POST")
 
+	router.Handle("/v1/keypairs/generate", Middleware(http.HandlerFunc(KeypairGenerateHandler))).Methods("POST")
+	router.Handle("/v1/keypairs/status/{authorityID}/{keyName}", Middleware(http.HandlerFunc(KeypairStatusHandler))).Methods("GET")
```

Comment:
Is it not needed here setting the valid pattern for the identifiers?

Reply:
These will get validate later. I can include something here, but it means that the rules may not match snapd's validation if that changes.